src/components/course/CourseEnrollmentFailedAlert.jsx

```diff
@@ -1,67 +1,129 @@
 import React, { useContext, useMemo } from 'react';
+import PropTypes from 'prop-types';
 import { useLocation } from 'react-router-dom';
 import { AppContext } from '@edx/frontend-platform/react';
 import { Container, Alert } from '@edx/paragon';
+import { Error } from '@edx/paragon/icons';
 
 import { useRenderContactHelpText } from '../../utils/hooks';
 import {
+  ENROLLMENT_COURSE_RUN_KEY_QUERY_PARAM,
   ENROLLMENT_FAILED_QUERY_PARAM,
   ENROLLMENT_FAILURE_REASON_QUERY_PARAM,
 } from './data/constants';
+import { CourseEnrollmentsContext } from '../dashboard/main-content/course-enrollments/CourseEnrollmentsContextProvider';
+
+export const ENROLLMENT_SOURCE = {
+  DASHBOARD: 'DASHBOARD',
+  COURSE_PAGE: 'COURSE_PAGE',
+};
+
+const createEnrollmentFailureMessages = (contactHelpText) => ({
+  dsc_denied: (
+    <>
+      You were not enrolled in your selected course. In order to enroll, you must accept the data sharing
+      consent terms. Please {contactHelpText} for further information.
+    </>
+  ),
+  verified_mode_unavailable: (
+    <>
+      You were not enrolled in your selected course as the verified course mode is
+      unavailable. Please {contactHelpText} for further information.
+    </>
+  ),
+  default: (
+    <>
+      You were not enrolled in your selected course. Please
+      {' '}{contactHelpText} for further information.
+    </>
+  ),
+});
+
+const createUpgradeFailureMessages = (contactHelpText, enrollmentSource) => ({
+  dsc_denied: (
```

Comment:
nit: dont think we necessarily should change this now, but food for thought... can we get the `failureMode` and these object keys to be camelCased and more JS-esque?

Reply:
we could camelcase the url params when we parse them.